sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/ViewRegistry.java

```diff
@@ -55,17 +62,42 @@ public static ViewRegistryBuilder builder() {
    * @param descriptor description of the instrument.
    * @return The {@link View} for this instrument, or a default aggregation view.
    */
-  public View findView(InstrumentDescriptor descriptor) {
-    LinkedHashMap<Pattern, View> configPerType = configuration.get(descriptor.getType());
-    for (Map.Entry<Pattern, View> entry : configPerType.entrySet()) {
-      if (entry.getKey().matcher(descriptor.getName()).matches()) {
+  public View findView(InstrumentDescriptor descriptor, InstrumentationLibraryInfo meter) {
+    for (Map.Entry<InstrumentSelector, View> entry : configuration.entrySet()) {
+      if (matchesSelector(entry.getKey(), descriptor, meter)) {
         return entry.getValue();
       }
     }
 
     return getDefaultSpecification(descriptor);
   }
 
+  // Matches an instrument selector against an instrument + meter.
+  private static boolean matchesSelector(
+      InstrumentSelector selector,
+      InstrumentDescriptor descriptor,
+      InstrumentationLibraryInfo meter) {
+    return (selector.getInstrumentType() == null
+            || selector.getInstrumentType() == descriptor.getType())
+        && matchesPattern(selector.getInstrumentNamePattern(), descriptor.getName())
+        && matchesMeter(selector.getMeterSelector(), meter);
+  }
+
+  // Matches a meter selector against a meter.
+  private static boolean matchesMeter(MeterSelector selector, InstrumentationLibraryInfo meter) {
+    return matchesPattern(selector.getNamePattern(), meter.getName())
+        && matchesPattern(selector.getVersionPattern(), meter.getVersion())
+        && matchesPattern(selector.getSchemaUrlPattern(), meter.getSchemaUrl());
+  }
+
+  // Matches a pattern against  a value.  Null values are treated as empty strings.
+  private static boolean matchesPattern(Pattern pattern, String value) {
+    if (value == null) {
+      return pattern.matcher("").matches();
```

Comment:
```suggestion
      value = "";
```

To not duplicate the logic